wandb/sdk/artifacts/storage_policies/wandb_storage_policy.py

```diff
@@ -144,9 +145,14 @@ def load_file(
                 http_headers["Authorization"] = f"Bearer {self._api.access_token}"
             elif _thread_local_api_settings.cookies is None:
                 auth = ("api", self._api.api_key or "")
-
             response = self._session.get(
-                self._file_url(self._api, artifact.entity, manifest_entry),
+                self._file_url(
+                    self._api,
+                    artifact.entity,
+                    artifact.project,
+                    artifact.name.split(":")[0],
```

Comment:
it is always guaranteed that we have the name field as a string, i didn't go take a look at the type definitions, just want to make sure that it can't be `None`, to avoid the split failing.

Reply:
same for project name is it always a string?

Reply:
Yup, `artifact.project` is always a string -> we assert that here: https://github.com/wandb/wandb/blob/bb0ac5d792a0fc8f9c37bd05cabbcd579f5994b4/wandb/sdk/artifacts/artifact.py#L477-L481

And `artifact.name` is also always a string. In fact, we use the same split to fetch the artifact's collection in a couple of places. https://github.com/wandb/wandb/blob/bb0ac5d792a0fc8f9c37bd05cabbcd579f5994b4/wandb/sdk/artifacts/artifact.py#L484
https://github.com/wandb/wandb/blob/bb0ac5d792a0fc8f9c37bd05cabbcd579f5994b4/wandb/sdk/artifacts/artifact.py#L515

Reply:
great, just wanted to make sure and like i said in my first message didn't check it myself so thank for verifying :)